[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64382 [ONNX] Deprecate use_external_data_format param from torch.onnx.export() function. (#62257)
* #64380 [ONNX] Deprecated the example_outputs param from torch.onnx.export() function. (#62815)
* #64373 [ONNX] minor doc improvements and cleanup (#62514)
* #64372 [ONNX] Add supplementary tests and description for custom_opsets param from torch.onnx.export() function. (#62085)
* #64371 [ONNX] Remove strip_doc_string param from torch.onnx.export() function. (#61712)
* **#64370 [ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)**

As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

Differential Revision: [D30905270](https://our.internmc.facebook.com/intern/diff/D30905270)

fatcat-z committed:
[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)

As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)"


As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)"


As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)"


As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30905270).

BowenBao committed:
Update on "[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)"


As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

Differential Revision: [D30905270](https://our.internmc.facebook.com/intern/diff/D30905270)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30905270).

BowenBao committed:
Update on "[ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)"


As of now, the "_retain_param_name" parameter has no description in PyTorch docs website. According to code, this argument determines if we keep the original parameter names of PyTorch model in the final ONNX graph. If this is False, those original parameter names will be replaced with a series of integers starting from 1.

Since setting numbers as parameter names make no sense to users, we remove this argument from the torch.onnx.export() function to increase user experience of calling this function.

This PR will still keep it in torch.onnx.export() function for backward support while all backend logic has been changed to work as _retain_param_name is set to True.

Co-authored-by: fatcat-z <zhang-ji@outlook.com>

Differential Revision: [D30905270](https://our.internmc.facebook.com/intern/diff/D30905270)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30905270).
